prime-router/src/main/kotlin/fhirengine/translation/hl7/FhirTransformer.kt

```diff
@@ -18,7 +20,7 @@ import org.hl7.fhir.r4.model.Extension
  */
 class FhirTransformer(
     private val schemaRef: FhirTransformSchema,
-) : ConfigSchemaProcessor() {
+) : ConfigSchemaProcessor<Bundle, Bundle, FhirTransformSchema, FhirTransformSchemaElement>(schemaRef) {
```

Comment:
I know we talked about this in standup, but can you please explain again why this was necessary so I can see the reason while being able to look through the code on my own?

Reply:
So the thought process was that for validation, every schema needs to be validated against some input and some output regardless of whether its FHIR->FHIR or FHIR->HL7 (and then even forward looking if we implemented schemas that were FHIR->CSV).  Since they all should do the same thing, convert an input using the schema and comparing it to an output, this logic should exist on `ConfigSchemaProcessor`.  In order to lift that into ConfigSchemaProcessor I needed to capture the types.

It has the side benefit of substantially cleaning up all the code as well and making it type safe.